xunit-autolabeler-v2/ast_parser/core/cli.py

```diff
@@ -36,12 +37,196 @@ def _write_output(output: List[str], output_file: str) -> None:
+    show_filenames: bool,
+    output_file: Optional[str] = None
+) -> None:
+    """Lists region tags in a directory.
+
+    This method lists the region tags for snippets parsed from a given
+    directory, filtered based on whether each tag was detected by an AST
+    parser (or not). It can also display test counts and/or source filenames
+    for each region tag.
+
+    Args:
+        data_json: A path to a polyglot_drift_data.json file for the specified
+                   root directory
+        root_dir: A path to the target root directory.
+        show_detected: Whether or not to show region tags that *were* detected
+                       by the AST parser
+        show_undetected: Whether or not to show region tags *not* detected by
+                         the AST parser
+        show_test_counts: Whether or not to show test counts for each
+                          AST-parser-detected region tag
+        show_filenames: Whether or not to show source filenames for each
+                        AST-parser-detected region tag
+        output_file: (Optional) A filepath to write the YAML validation
+                     results to. Results will be written to stdout if this
+                     argument is omitted.
+    """
+    output = []
+
+    if show_undetected and show_test_counts:
+        output.append(
+            'WARN Undetected/ignored region tags do not have test counts')
+
+    grep_tags, source_tags, ignored_tags, source_methods = (
+        analyze.analyze_json(data_json, root_dir))
+
+    def _get_test_count_str(region_tag):
+        if not show_test_counts:
+            return ''
+
+        test_data_matches = [method for method in source_methods if
+                             region_tag in method['region_tags']]
+
+        total_tests = 0
+        for test_data in test_data_matches:
+            total_tests += len(test_data['test_methods'])
+
+        return f'({total_tests} test(s))'
+
+    if show_detected:
+        output.append('Detected region tags:')
+        for tag in source_tags:
+            output.append(f'  {tag} {_get_test_count_str(tag)}')
+            if show_filenames:
+                source_file = [method['source_path']
+                               for method in source_methods
+                               if tag in method['region_tags']][0]
+                output.append(f'    Source file: {source_file}')
+
+    if show_undetected:
+        output.append('Undetected region tags:')
+        undetected_tags = [tag for tag in grep_tags
+                           if tag not in source_tags]
+        undetected_tags = [tag for tag in undetected_tags
+                           if tag not in ignored_tags]
+        for tag in undetected_tags:
+            output.append(f'  {tag}')
+
+    if ignored_tags:
+        output.append('Ignored region tags')
+        for tag in ignored_tags:
+            output.append(f'  {tag}')
+
+    _write_output(output, output_file)
+
+
+def list_source_files(
```

Comment:
This function too. Can we have a function that returns source file paths as a list of strings?
I think that will clarify the intention of the function better.

Reply:
Ditto to above - we'd have to send a rendering method __both__ `untested_files` **and** `tested_files`.

Reply:
Refactored these into separate files; marking this as resolved.